Feature/Routing-Forms/Reporting

hariombalhara opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Closes #5440

### [Demo](https://www.loom.com/share/b2cd9f83b0ea418a95804c20f90ab090)
<img width="1189" alt="Screenshot 2022-11-09 at 9 13 48 PM" src="https://user-images.githubusercontent.com/1780212/200875215-9b4e71ad-2907-414f-bfbd-c3f7b577501c.png">

**Environment**: Production

## Type of change

<!-- Please delete options that are not relevant. -->

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- See loom



_Note: You would see 2 unit tests check below. One is passing and the other is failing. This is because I have a fix for unit tests workflow in my PR which is passing and `main` one is failing_

hariombalhara committed:
Add very basic reporting logic

hariombalhara committed:
Merge remote-tracking branch 'origin/main' into feat/routing-forms/reporting

harshsinghatz commented on packages/app-store/ee/routing-forms/jsonLogicToPrisma.ts:
```diff
@@ -0,0 +1,25 @@
+export const jsonLogicToPrisma = (jsonLogic) => {
```
what is the type for jsonLogic?

harshsinghatz replied:
what are the possible types for jsonLogic here?

harshsinghatz commented on packages/app-store/ee/routing-forms/jsonLogicToPrisma.ts:
```diff
@@ -0,0 +1,25 @@
+  if (!jsonLogic) {
+    return {};
+  }
+  jsonLogic = jsonLogic.logic;
```
Can we store this in a different variable, as the parameter is directly getting manipulated

hariombalhara commented:
@harshsinghatz this isn't ready for review yet. It's in draft

hariombalhara committed:
Merge remote-tracking branch 'origin/main' into feat/routing-forms/reporting

hariombalhara committed:
Support more logical operators

hariombalhara committed:
Add tests

hariombalhara committed:
Merge remote-tracking branch 'origin/main' into feat/routing-forms/reporting

hariombalhara committed:
Fix types

hariombalhara committed:
Merge remote-tracking branch 'origin/main' into feat/routing-forms/reporting

hariombalhara committed:
Merge remote-tracking branch 'origin/main' into feat/routing-forms/reporting

hariombalhara committed:
Add playwright test for reporting

hariombalhara committed:
Fix table UI

hariombalhara committed:
Remove only

hariombalhara committed:
Better name variable

hariombalhara committed:
Increase max_old_space to allow tests to run without reaciing limit

hariombalhara committed:
Merge remote-tracking branch 'origin/main' into feat/routing-forms/reporting

hariombalhara committed:
Update test name

hariombalhara committed:
Reuse getServerSideProps

hariombalhara committed:
Variable renaming

hariombalhara reviewed (commented):
Self review done

hariombalhara committed:
Merge remote-tracking branch 'origin/main' into feat/routing-forms/reporting

CarinaWolli commented:
Awesome feature, works great 🙌🏻 

I only have a few design suggestions: 

- There is more space on the right side of table (I think there is a mr-2 somewhere)
![Screenshot 2022-11-10 at 10 11 23](https://user-images.githubusercontent.com/30310907/201072117-27c15dff-5691-4245-94e7-4900a83fc33b.png)

- Pagination would probably make sense to have as there will be a lot of data over time (maybe this can also wait and be added later)
- Should we make the corners of the table rounded to fit our design?

PeerRich reviewed (approved):
the gray headline looks a bit dark, a lighter dark is better

also "Add Query" should be "Add Filter", no?

hariombalhara committed:
Merge remote-tracking branch 'origin/main' into feat/routing-forms/reporting

hariombalhara committed:
PR Suggestions

hariombalhara commented:
@PeerRich 
- Changed Add Query ->  Add Filter. It would resonate with more people.
- Made header bg-gray-200
<img width="1174" alt="Screenshot 2022-11-11 at 11 57 07 AM" src="https://user-images.githubusercontent.com/1780212/201277337-9394f6a1-301f-41aa-b48a-5f9c269cf913.png">

@CarinaWolli 
- "There is more space on the right side of table (I think there is a mr-2 somewhere)" -> This is because the table is left aligned. The space at right would vary 
- "Pagination would probably make sense to have as there will be a lot of data over time (maybe this can also wait and be added later)" -> Yeah I thought the same. The user can always filter the data to limit what loads on the page
- "Should we make the corners of the table rounded to fit our design?" - Wanted to do it but it wasn't straightforward.  Tables are tricky. Implemented it now.

kodiakhq[bot] committed:
Merge branch 'main' into feat/routing-forms/reporting

kodiakhq[bot] committed:
Merge branch 'main' into feat/routing-forms/reporting
